src/execution/sql/operators/like_operators.cpp

```diff
@@ -75,7 +75,9 @@ bool Like::Impl(const char *str, size_t str_len, const char *pattern, size_t pat
       return false;
     }
   }
-
+  while (plen > 0 && *p == '%') {
```

Comment:
What does this do? The loop condition above was `plen > 0 && slen > 0`, so for us to have reached here, it must be the case that `plen <= 0 || slen <= 0`. Given your condition here, this executes when `slen <= 0`. Your code advances through the rest of the pattern, so if the string runs out of characters right as the pattern reaches a %, you will always say the string fits the pattern.

I am not convinced that this is correct, consider pattern `terrier%groundhog` and string `terrier`. What does your code currently do for that case?

Reply:
I am mistaken, LGTM.